# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: third_party/rust/ash/src/lib.rs
# Commit: ef5dc3e04e5f
# Full Hash: ef5dc3e04e5f271eea0636ab3a495e95cc912f1d
# Author: Dzmitry Malyshau <dmalyshau@mozilla.com>
# Date: 2021-09-04 09:40:24
# Regressor Bug: 1726626
# File Overlap Count: 3
# Description:
#   Bug 1726626 - Move gfx/wgpu into a 3rd party dependency r=jgilbert,bholley
#   
#   This update makes wgpu a vendored dependency instead of having it in gfx/wgpu.
#   
#   ## Notes
# ==============================================================================

diff --git a/third_party/rust/ash/src/lib.rs b/third_party/rust/ash/src/lib.rs
--- a/third_party/rust/ash/src/lib.rs
+++ b/third_party/rust/ash/src/lib.rs
@@ -4,51 +4,58 @@
     clippy::upper_case_acronyms
 )]
 //! # Vulkan API
 //!
 //! <https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/index.html>
 //!
 //! ## Examples
 //!
-//! ```rust,no_run
-//! use ash::{vk, Entry, version::EntryV1_0};
+//! ```no_run
+//! use ash::{vk, Entry};
 //! # fn main() -> Result<(), Box<dyn std::error::Error>> {
-//! let entry = Entry::new()?;
+//! let entry = unsafe { Entry::new() }?;
 //! let app_info = vk::ApplicationInfo {
-//!     api_version: vk::make_version(1, 0, 0),
+//!     api_version: vk::make_api_version(0, 1, 0, 0),
 //!     ..Default::default()
 //! };
 //! let create_info = vk::InstanceCreateInfo {
 //!     p_application_info: &app_info,
 //!     ..Default::default()
 //! };
 //! let instance = unsafe { entry.create_instance(&create_info, None)? };
 //! # Ok(()) }
 //! ```
 //!
+//! ## Getting started
+//! Load the Vulkan library at the default location using [`Entry::new()`][EntryCustom<_>::new()],
+//! or at a custom location using [`Entry::with_library("path/to/vulkan")`][EntryCustom<_>::with_library()].
+//! These loaders use [`libloading`]. If you wish to perform function loading yourself
+//! call [`EntryCustom::new_custom()`] with a closure turning function names
+//! into function pointers.
 
 pub use crate::device::Device;
 pub use crate::entry::{EntryCustom, InstanceError};
 #[cfg(feature = "libloading")]
 pub use crate::entry_libloading::{Entry, LoadingError};
 pub use crate::instance::Instance;
 
 mod device;
 mod entry;
 #[cfg(feature = "libloading")]
 mod entry_libloading;
 mod instance;
 pub mod prelude;
 pub mod util;
-pub mod version;
+/// Raw Vulkan bindings and types, generated from `vk.xml`
 #[macro_use]
 pub mod vk;
 
 // macros of vk need to be defined beforehand
+/// Wrappers for Vulkan extensions
 pub mod extensions;
 
 pub trait RawPtr<T> {
     fn as_raw_ptr(&self) -> *const T;
 }
 
 impl<'r, T> RawPtr<T> for Option<&'r T> {
     fn as_raw_ptr(&self) -> *const T {